src/foundry/foundry.js/applications/filePicker.d.ts

```diff
@@ -81,7 +92,7 @@ declare class FilePicker<P extends FilePicker.Options = FilePicker.Options> exte
    * Parse a s3 key to learn the bucket and the key prefix used for the request
    * @param key - A fully qualified key name or prefix path
    */
-  protected static parseS3URL(key: string): { bucket: string; keyPrefix: string };
+  static parseS3URL(key: string): { bucket: string | null; keyPrefix: string };
 
   /**
    * @override
```

Comment:
For the methods below:

- `browse()`:
    Please document the default value of the `options` parameter
    Return type should be `Promise<FilePicker.Result | undefined>`
- `static browse()`:
    Please document the default value of the `options` parameter
- `configurePath()`:
    Please document the default value of the `options` parameter
- `createDirectory()`:
    Please document the default value of the `options` parameter
- `upload()`:
    Please document the default value of the `options` parameter
- `render()`:
    I guess the return type should be something like `void | ReturnType<this['browse']>`
- `_onRequestTarget()`:
    I guess the return type should be something like `void | ReturnType<this['browse']>`
- `_onLazyLoadImages()`:
    Parameters can be grabbed from `SidebarTab#_onLazyLoadImages`

Reply:
@UFOMelkor Work is in progress. I have to say, most of these things are not 0.8.x-related changes but I'm happy to help get the quality level up on the exsiting stuff.

When you say "document the default value of the options parameter", is it sufficient to document to default values of the relvent options type? Or is there some `/** */` magic to add a default value to a function arg?

Reply:
Normally, we do it like this:
```
   * @param options - Additional file upload options passed as form data
   *                  (default: `{}`)
```

>  but I'm happy to help get the quality level up on the exsiting stuff.

Happy to hear :-)

Reply:
Ohhh I see. That's easy enough. I was trying to cram `@default` in there :)

Reply:
To clarify this a bit:

Often there a functions like this

```javascript
/**
 * @param {object} [options] Additional options
 * @param {string} [options.bar=baz] Some description
 */
function foo({bar = 'baz'} = {}) {}
```

and we document it like this.

```typescript
/**
 * @param options - Additional options
 *                  (default: `{}`)
 */
function foo(options?: Partial<Options>): void;

interface Options {
    /**
     * Some description
     * @defaultValue `'baz'`
     */
    bar: string;
}

Reply:
So I think we're ready for another review. All the previous items have been addressed, plus I've added more options types to replace those which were previously `Record<string, any>`.

I've also broken up the result types, so `BrowseResult` is different to `ConfigurePathResult`.

Finally, and this may be the controversial bit - I've created a helper type called `PromisedReturnType` which helps when you're forwarding the result of a promise. There are other ways of doing this, including just hard-coding the promised type, but this feels like the most correct solution. If I have over-engineered this just for sake of making `_onDrop` cleaner I will revert those changes :)